notes/books/algorithms-live-by.md

```diff
@@ -168,3 +168,66 @@ tired or slow, the path to get there got a lot bigger =).
 As usual it is a trade-off, if you know
 more it will be slower to navigate through everything that
 you know, knowing less makes you look really fast and sharp.
+
+# Data Idolatry and Metrics
+
+There is a great analogy of idolatry in the old days and
+how we look at data today. He talks a little about how god
+both in the Old and the New testament was always pissed off
+with the fact that people would give devotion to what they
+can see, like the wooden and metal idols, instead of giving
+it to god, which is so awesome that nothing that can be
+seen can represent him properly.
+
+What we do today with data is the same thing. Any data or
+metric that we can gather is always a simpler representation
+from reality. The idea is not that you should not do that,
+or not try to improve that, but to forget what the data and
+the metric really is, just a simpler/imperfect representation
+of a much more complex phenomenon. Focusing too much
+on this representation can lead to problems, which the most
+common one is over fitting.
+
+Over fitting is a well know phenomenon know by statisticians,
+but it is very interesting how it also happens with humans
+when analyzed as a system. One human example is eating habits
+and health. Our brain has a builtin metric/representation
+of what we should eat, that is taste. Taste optimizes for what
+we need most, and usually it is energy. So our taste optimizes
+for high energy food, it is the best fit. But once humans started
+to control the nutritional components of their food they immediately
+went to over fitting, food is now extremely biased towards our
+metric (taste) and guess what, as all metrics it is not a perfect
+measurement of nutritious value, hence if you over fit it you
+end up not healthy. This metric worked because no food had
+great amounts of sugar in it in nature, as soon as we could
+control that the metric was not accurate anymore (it never was
+actually) and we went to full over fitting in a single generation.
+
+After building up this awesome analogy he goes on with a lot
+of examples on different industries of when you establish a
+metric people over fit to it in a way that stops being beneficial.
+One example is a law firm that measures closed cases, people in
+the end of the month go to the smaller non important cases
+to boost to the metric.
+
+The same goes for closed issues/bugs in software projects.
+You can enrich your metrics, but
+it seems to me that extreme devotion to any metric, even if it is
+rich, can always lead to some sort of disappointment. It is
+better to have the metric but at the same time don't idolize it
+as the only source of truth.
```

Comment:
This reminds me of some stupid ideas people talk about, like ranking software over lines of code. Or ranking organizations over contributions to Github, number of repositories, etc.

Reply:
Yeah, when you can't measure what is important you will make what you can measure important x_x (and some things are extremely hard to measure)